core/genesis/file.go

```diff
@@ -0,0 +1,253 @@
+package genesis
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"google.golang.org/protobuf/proto"
+)
+
+var _ GenesisSource = &FileGenesisSource{}
+
+type FileGenesisSource struct {
+	sourceDir  string
+	moduleName string
+}
+
+const (
+	fileOpenflag  = os.O_CREATE | os.O_WRONLY
+	flieOpenMode  = fs.FileMode(0o600)
+	dirCreateMode = fs.FileMode(0o700)
+)
+
+// NewFileGenesisSource returns a new GenesisSource for the provided
+// source directory and the provided module name where it is assumed
+// that the encoded json data of file.
+func NewFileGenesisSource(sourceDir, moduleName string) *FileGenesisSource {
```

Comment:
If all fields are private, why are we not returning the interface?

```suggestion
func NewFileGenesisSource(sourceDir, moduleName string) GenesisSource {
```

Reply:
thanks for the review. I forgot the turn it to the interface after I remove some public functions don't need.